sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/CustomModelTransforms.java

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.formrecognizer;
+
+import com.azure.ai.formrecognizer.implementation.models.ErrorInformation;
+import com.azure.ai.formrecognizer.implementation.models.Model;
+import com.azure.ai.formrecognizer.implementation.models.ModelStatus;
+import com.azure.ai.formrecognizer.models.CustomFormModel;
+import com.azure.ai.formrecognizer.models.CustomFormModelField;
+import com.azure.ai.formrecognizer.models.CustomFormSubModel;
+import com.azure.ai.formrecognizer.models.FormRecognizerError;
+import com.azure.ai.formrecognizer.models.ModelTrainingStatus;
+import com.azure.ai.formrecognizer.models.TrainingDocumentInfo;
+import com.azure.ai.formrecognizer.models.TrainingStatus;
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Helper class to convert service level custom form related models to SDK exposed models.
+ */
+final class CustomModelTransforms {
+    private static final ClientLogger LOGGER = new ClientLogger(CustomModelTransforms.class);
+
+    private CustomModelTransforms() {
+    }
+
+    static CustomFormModel toCustomFormModel(Model modelResponse) {
+        com.azure.ai.formrecognizer.implementation.models.ModelInfo modelInfo = modelResponse.getModelInfo();
+        if (modelInfo.getStatus() == ModelStatus.INVALID) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException("Invalid status Model Id."));
+        }
+
+        List<FormRecognizerError> trainResultErrors = new ArrayList<>();
+        if (modelResponse.getTrainResult().getErrors() != null) {
+            trainResultErrors = setTrainingErrors(modelResponse.getTrainResult().getErrors());
+        }
```

Comment:
Could `setTrainingErrors` `null` check the passed `List` so we don't need to maintain a new `List` outside of the scope of the `if`?